packages/codegen-ui/lib/utils/form-to-component/map-form-definition-to-component.ts

```diff
@@ -98,55 +98,84 @@ const ctaButtonConfig = (): StudioComponentChild => {
         value: '1rem',
       },
     },
-    children: [
-      {
-        componentType: 'Button',
-        name: 'CancelButton',
-        properties: {
-          children: {
-            value: 'Cancel',
-          },
-          type: {
-            value: 'button',
-          },
+    children: [],
+  };
+
+  if (formDefinition.buttons.buttonConfigs.cancel) {
+    CTAComponent.children?.push({
+      componentType: 'Button',
+      name: 'CancelButton',
+      properties: {
+        children: {
+          value: formDefinition.buttons.buttonConfigs.cancel.props.children,
```

Comment:
you could create a helper that maps buttons instead of doing it one by one. Something like

```
mapButton(buttonElement, buttonName // this would be the key you get when Object.entries) {
  return {componentType: element.componentType,
name: transformIntoButtonname(buttonName),
properties: mapFormElementProps(element) // you can map all the props like type and variation (not variant like you have now. my bad) in the form definition mapper all you gotta do is just map them on
```